### PR TITLE
DS-735 | @ericbakenhus | Event Cards

### PR DESCRIPTION
**Pull request by ericbakenhus**
# READY FOR REVIEW

# Summary
- Adds an event card list block
- Minor style tweaks for events

# Review By (Date)
- EOD 9/13

# Criticality
- Medium

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. Navigate to /test-eric/test-basic/
3. Verify that event cards are shown
4. Optionally, edit the event card list options in the SB dev space and verify that changes flow through as expected (after restarting your local instance)

# Associated Issues and/or People
- DS-735

### CODE REVIEW

File: src/components/content-types/synchronized-event/synchronizedEventCardList.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+          <SynchronizedEventCard key={event.id} {...event.content} />
+        ))}
+        {hasLastCard && <CreateBloks blokSection={lastCard} />}
+      </div>
+    </SbEditable>
+  );
+};
+SynchronizedEventCardList.propTypes = SynchronizedEventCardListProps;
+
+export default SynchronizedEventCardList;
```
Comment (ericbakenhus) [inline src/components/content-types/synchronized-event/synchronizedEventCardList.js:56]: New sync event list block component. Displays a number of events based on selected options.

File: src/components/content-types/synchronized-event/synchronizedEventCard.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+          eventTimezone={eventTimezone}
+          location={location}
+          city={city}
+          region={region}
+          subject={subject}
+        />
+      </div>
+    </div>
+  );
+};
```
Comment (ericbakenhus) [inline src/components/content-types/synchronized-event/synchronizedEventCard.js:95]: Internal sync event card component. Displays an event as a card. Only used by `synchronizedEventCardList`.

File: src/components/events-discovery/components/Event/EventContent.jsx [hunk 3]
```diff
@@ -1,13 +1,12 @@
...
+import { MenuItem, Select } from '@mui/material';
 import {
   CalendarIcon,
   LocationMarkerIcon,
-  TagIcon,
-} from '@heroicons/react/solid';
-import { GlobeIcon } from '@heroicons/react/outline';
-import { MenuItem, Select } from '@mui/material';
+  GlobeIcon,
+} from '@heroicons/react/outline';
```
Comment (ericbakenhus) [inline src/components/events-discovery/components/Event/EventContent.jsx:9]: Just swapping out the icons for their outline variants.

File: src/components/events-discovery/components/Event/EventContent.jsx [hunk 4]
```diff
@@ -186,8 +185,7 @@ export const EventContent = ({
         </div>
       )}
       {!!subject?.length && (
-        <div className="su-flex su-items-center su-mb-04em sm:su--ml-30 su-mt-20">
-          <TagIcon className={iconClasses} aria-hidden="true" />
+        <div className="su-flex su-items-center su-mb-04em su-mt-20">
```
Comment (ericbakenhus) [inline src/components/events-discovery/components/Event/EventContent.jsx:188]: Jettisoning the tag icon.

**Comment by sherakama** [comment]
Screenshot 2024-09-05 at 14 27 59

Screenshot 2024-09-05 at 14 27 42

Is the red card supposed to be that wide on mobile? Sticks out.

File: src/components/content-types/synchronized-event/synchronizedEventCardList.js
Comment (sherakama) [inline src/components/content-types/synchronized-event/synchronizedEventCardList.js:1]: Missing the `x` in `jsx`

File: src/components/content-types/synchronized-event/synchronizedEventCardList.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  filename: PropTypes.string,
+  focus: PropTypes.string,
+  imageFocus: PropTypes.string,
+  tabText: PropTypes.string,
+  headingLevel: HeadingLevelType,
+  orientation: PropTypes.string,
+  spacingBottom: PropTypes.string,
+  isDark: PropTypes.bool,
+  className: ClassNameType,
+};
```
Comment (sherakama) [inline src/components/content-types/synchronized-event/synchronizedEventCardList.js:24]: Why did you choose the js proptypes here but have been going with the docbloc comments elsewhere?

File: src/hooks/useEvents.js [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+        formats?.length &&
+        !formats.some((s) => event.content?.format?.includes(s))
+      ) {
+        return false;
+      }
+
+      return true;
+    })
+    .slice(0, maxNumEvents);
+};
```
[the remark this answers is not in the document]
Comment (sherakama) [reply src/hooks/useEvents.js:92]: Neat.

File: src/components/content-types/synchronized-event/synchronizedEventCardList.js (hunk 5 above)
Comment (ericbakenhus) [reply src/components/content-types/synchronized-event/synchronizedEventCardList.js:24]: I used JSDoc to help me with the intellisense for all the internal components within the event discovery stuff. For this block I tried to follow the conventions being used by other SB block components.

File: src/components/content-types/synchronized-event/synchronizedEventCardList.js
Comment (ericbakenhus) [reply src/components/content-types/synchronized-event/synchronizedEventCardList.js:1]: I blame peer pressure. All the existing SB block components are using `.js`. 😬

**Comment by sherakama** [comment]
Good options in SB too.

Screenshot 2024-09-05 at 14 40 55
Screenshot 2024-09-05 at 14 40 39

**Comment by ericbakenhus** [comment]
> Is the red card supposed to be that wide on mobile? Sticks out.

@sherakama I stole that card from the current home page (it's just a CTA card block). That does seem to be what it does currently:

Screenshot 2024-09-05 at 4 43 08 PM

**Comment by ericbakenhus** [comment]
> Good options in SB too.

@sherakama The `Event Card (Vertical)` is the existing event card. I don't know what they are planning to do with it, but I'm assuming it's getting tossed along with the existing (non-synchronized) event stories.

**Comment by sherakama** [comment]
We should add a ticket for that and ask to find out.

File: src/components/content-types/synchronized-event/synchronizedEventCardList.js (hunk 5 above)
Comment (ericbakenhus) [reply src/components/content-types/synchronized-event/synchronizedEventCardList.js:24]: Also, I completely forgot that it existed. Probably needs some updates...

**Comment by ericbakenhus** [comment]
> We should add a ticket for that and ask to find out.

It looks some card blocks apply a max width at that screen size and some don't. I'll let Julianne know about the discrepancy to see what they want to do.

File: src/components/content-types/synchronized-event/synchronizedEventCardList.js
Comment (ericbakenhus) [reply src/components/content-types/synchronized-event/synchronizedEventCardList.js:1]: It looks like the previous conventions was to use camel case and `.js` for SB block components and pascal case and `.jsx` for internal components. Do we want to continue with that (until the Next.js rebuild)?

Comment (ericbakenhus) [reply src/components/content-types/synchronized-event/synchronizedEventCardList.js:1]: I went ahead and swapped the three components in `content-types/synchronized-event` to use `.jsx`. I switched the one internal component to pascal case, but left the to block components as camel case. 🤷

File: src/components/content-types/synchronized-event/synchronizedEventCardList.js (hunk 5 above)
Comment (ericbakenhus) [reply src/components/content-types/synchronized-event/synchronizedEventCardList.js]: I jettisoned it.

**Comment by ericbakenhus** [comment]
@sherakama We have a new deadline for this. The full public launch will be a bit later.

**Comment by ericbakenhus** [comment]
> We should add a ticket for that and ask to find out.

@sherakama I've let Julianne know. She's checking with the client. "Our" recommendation was to apply the same existing max width to all card blocks.

**Comment by ericbakenhus** [comment]
> > We should add a ticket for that and ask to find out.
> 
> @sherakama I've let Julianne know. She's checking with the client. "Our" recommendation was to apply the same existing max width to all card blocks.

@sherakama Yep, we're going with adding the same max width to all the cards. I'll spin up a new ticket for it. 👍

**Comment by sherakama** [comment]
> @sherakama I'm going to move this to dev so the client can begin QA. Please feel free to retro.

All good. Thanks.

**Review by sherakama — COMMENTED** [review]
GTG

**Pull request merged**